Cricket Scoreboard/Score.swift

```diff
@@ -8,139 +8,180 @@
 
 import Foundation
 
-class Score: NSObject, XMLParserDelegate {
+private extension Array {
+    subscript (safe index: Int) -> Element? {
+        return indices ~= index ? self[index] : nil
+    }
+}
+
+private enum ScoreElement: String {
+    case title, guid
+}
+
+struct ScoreItem {
+    let title: String
+    let link: String
+}
+
+class Score: NSObject {
     
     // Map storing emoji flags for each country
-    let flags = ["aus": "\u{1F1E6}\u{1F1FA}", "ind": "\u{1F1EE}\u{1F1F3}", "sa": "\u{1F1FF}\u{1F1E6}", "nz": "\u{1F1F3}\u{1F1FF}", "sl":"\u{1F1F1}\u{1F1F0}", "eng": "\u{1F1EC}\u{1F1E7}", "ban": "\u{1F1E7}\u{1F1E9}", "pak":"\u{1F1F5}\u{1F1F0}", "wi":"", "ire": "\u{1F1EE}\u{1F1EA}", "zim": "\u{1F1FF}\u{1F1FC}", "afg": "\u{1F1E6}\u{1F1EB}"]
-    
+    private let flags = ["aus": "🇦🇺", "ind": "🇮🇳", "sa": "🇿🇦",
+                         "nz": "🇳🇿", "sl": "🇱🇰", "eng": "🏴󠁧󠁢󠁥󠁮󠁧󠁿",
+                         "ban": "🇧🇩", "pak": "🇵🇰", "wi": "",
+                         "ire": "🇮🇪", "zim": "🇿🇼", "afg": "🇦🇫"]
+
+    private let liveScoresUrl =
+        URL(string: "http://static.cricinfo.com/rss/livescores.xml")!
+    private let matchListUpdateInterval = 25.0
+
     // By default first match in the list is selected
     var selectedMatch = 0
-    let matchListUpdateInterval = 25.0
-    
-    // Cricinfo RSS Link
-//    let RSS_URL = "http://static.cricinfo.com/rss/livescores.xml"
-    
-    var parser = XMLParser()
-    var posts = [[String: String]]()
-    var elements = [String: String]()
-    var element = NSString()
-    var title1 = NSMutableString()
-    var link = NSMutableString()
-    
-    var url = URL(string: "http://static.cricinfo.com/rss/livescores.xml")
-    var onUpdateListener: (String, [[String: String]]) -> Void
+
+    private var allMatchScores = [ScoreItem]()
+    private var currentScoreElement: ScoreElement?
+    private var currentTitle = ""
+    private var currentLink = ""
+
+    typealias ScoreUpdate = (String, [ScoreItem]) -> Void
+    var onUpdateListener: ScoreUpdate
     
-    init(onUpdateListener: @escaping (String, [[String: String]]) -> Void) {
+    init(onUpdateListener: @escaping ScoreUpdate) {
         self.onUpdateListener = onUpdateListener
         super.init()
-        posts = []
-//        url = NSURL(string: RSS_URL)!
-        Timer.scheduledTimer(timeInterval: matchListUpdateInterval, target: self, selector: #selector(updateScore), userInfo: nil, repeats: true)
+
+        Timer.scheduledTimer(timeInterval: matchListUpdateInterval,
```

Comment:
.fire() at the end will always fire the timer at start , therefore no need to call updateScore manually from AppDelegate